internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -291,7 +291,7 @@ func (r *Reconciler) reconcileMachineHealthCheck(ctx context.Context, current, d
 	ctx, log = log.WithObject(current).Into(ctx)
 
 	// Check differences between current and desired MachineHealthChecks, and patch if required.
-	patchHelper, err := mergepatch.NewHelper(current, desired, r.Client)
+	patchHelper, err := mergepatch.NewHelper(current, desired, r.Client, mergepatch.AuthoritativePaths{contract.Path{"spec"}})
```

Comment:
nit: let's explain why we are doing this
```suggestion
	// NOTE: we want to be authoritative on the entire spec because the users are
	// expected to change MHC fields from the ClusterClass only.
	patchHelper, err := mergepatch.NewHelper(current, desired, r.Client, mergepatch.AuthoritativePaths{contract.Path{"spec"}})
```

Reply:
Just to surface it here too.

> It kind of works. So the patch helper unsets the field correctly. Unfortunately, because of the diff between the MHC in the ClusterClass (fields unset) and the defaulted fields in the MHC it permanently tries to patch the MHC
> https://kubernetes.slack.com/archives/C8TSNPY4T/p1642505125061200?thread_ts=1642501392.050900&cid=C8TSNPY4T

Reply:
I'll continue further discussion on the issue